### PR TITLE
Bluetooth: host: Split SSP to its own module

### PR DESCRIPTION
**Pull request by joerchan**
Refactor the BR/EDR SSP feature out into it's own source file.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

File: subsys/bluetooth/host/hci_core.c [hunk 1]
```diff
@@ -5208,23 +4872,24 @@ static const struct event_handler normal_events[] = {
 		      sizeof(struct bt_hci_evt_conn_request)),
 	EVENT_HANDLER(BT_HCI_EVT_CONN_COMPLETE, conn_complete,
```
Comment (joerchan) [inline subsys/bluetooth/host/hci_core.c:4873]: @jhedberg I was thinking of continuing the pattern below to move other parts out, for example conn_complete here can be moved to conn.c as well as other conn related events. And adv/scan can be similarly extracted, although I would like periodic advertising in first. Do you see this as a good way forward?

Comment (jhedberg) [reply subsys/bluetooth/host/hci_core.c:4873]: Sure, sounds like a good plan

**Comment by joerchan** [comment]
DNM: Needs to be updated and merged after https://github.com/zephyrproject-rtos/zephyr/pull/27243

File: subsys/bluetooth/host/hci_core.c (hunk 1 above)
Comment (carlescufi) [reply subsys/bluetooth/host/hci_core.c:4805]: Yes, this makes sense, +1 to that.

**Review by carlescufi — APPROVED** [review]
Fine with this, LGTM

**Comment by joerchan** [comment]
> DNM: Needs to be updated and merged after #27243

Rebased.

**Comment by joerchan** [comment]
@jhedberg I made same changes, would you mind looking at it one more time?

**Review by jhedberg — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**